pkg/api/types.go

```diff
@@ -313,6 +313,8 @@ const (
 type PersistentVolumeStatus struct {
 	// Phase indicates if a volume is available, bound to a claim, or released by a claim
 	Phase PersistentVolumePhase `json:"phase,omitempty"`
+	// Current service state of volume
+	Conditions []PersistentVolumeCondition `json:"conditions,omitempty" patchStrategy:"merge" patchMergeKey:"type"`
```

Comment:
As for two conditions of the same type ... My current usage of the conditions is to keep them a distinct set.  The controller code appends them when not there, else sets the values on the existing condition.